web/html/src/manager/virtualization/guests/list/guests-list.js

```diff
@@ -0,0 +1,401 @@
+const React = require("react");
+const PropTypes = React.PropTypes;
+const MessageContainer = require("../../../../components/messages").Messages;
+const MessagesUtils = require("../../../../components/messages").Utils;
+const {Table, Column, SearchField, Highlight} = require("../../../../components/table");
+const Functions = require("../../../../utils/functions");
+const Utils = Functions.Utils;
+const {LinkButton, AsyncButton} = require("../../../../components/buttons");
+const {DeleteDialog} = require("../../../../components/dialog/DeleteDialog");
+const {DangerDialog} = require("../../../../components/dialog/DangerDialog");
+const {ModalButton} = require("../../../../components/dialog/ModalButton");
+const Systems = require("../../../../components/systems");
+import VirtualizationGuestActionApi from '../virtualization-guest-action-api';
+import VirtualizationGuestsListRefreshApi from '../virtualization-guests-list-refresh-api';
+
+class GuestsList extends React.Component {
+
+  constructor(props) {
+    super(props);
+
+    this.state = {
+      actionsResults: {},
+      errors: [],
+      selectedItems: []
+    }
+
+    this.onBeforeUnload = (event) => {
+        this.setState({
+            pageUnloading: true
+        });
+    };
+
+    this.handleSelectItems = (items) => {
+      this.setState({
+        selectedItems: items
+      });
+    };
+
+    this.selectGuest = (row) => {
+      this.setState({
+        selected: row
+      });
+    };
+
+    this.actionCallback = (results) => {
+      var actions = this.state.actionsResults;
+      Object.entries(results).forEach(([uuid, actionid]) => {
+        actions[uuid] = {'id': actionid, 'result': 'Queued'};
+      });
+      this.setState({actionsResults: actions});
+    };
+  }
+
+  componentDidMount() {
+    if (this.props.salt_entitled) {
+      var port = window.location.port;
```

Comment:
nitpicking: we should start using const + let instead of var, this is just a point for the future it doesn't block the PR